pkg/cmd/dockerregistry/dockerregistry.go

```diff
@@ -42,19 +42,22 @@ import (
 	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
 	"github.com/openshift/origin/pkg/dockerregistry/server"
 	"github.com/openshift/origin/pkg/dockerregistry/server/audit"
+	registryconfig "github.com/openshift/origin/pkg/dockerregistry/server/configuration"
+	"github.com/openshift/origin/pkg/dockerregistry/server/metrics"
 )
 
 // Execute runs the Docker registry.
 func Execute(configFile io.Reader) {
-	config, err := configuration.Parse(configFile)
+	dockerConfig, extraConfig, err := registryconfig.Parse(configFile)
 	if err != nil {
 		log.Fatalf("error parsing configuration file: %s", err)
 	}
-	setDefaultMiddleware(config)
-	setDefaultLogParameters(config)
+	setDefaultMiddleware(dockerConfig)
+	setDefaultLogParameters(dockerConfig)
 
 	ctx := context.Background()
-	ctx, err = configureLogging(ctx, config)
+	ctx = server.WithConfiguration(ctx, extraConfig)
```

Comment:
I think passing variables through a context is a code smell, like a having global variables. It's very convenient to write this way (you can get data from anywhere), but it really complicates the code.

But in this case I don't care, I want to try to refactor the `repository` struct, after which there can emerge a place to pass the config. But it's too scary to do that without the good test coverage.